doc/design.adoc

```diff
@@ -0,0 +1,637 @@
+|
+
+| `secret`
+| Secret to connect to remote site (only for `Dynamic` type).
+| <<infinispansecretspec>>
+| false
+|
+
+|=======================
+
+
+[[infinispansecretspec]]
+##### `InfinispanSecretSpec`
+
+`InfinispanSecretSpec` defines how `InfinispanSpec` secrets are configured.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required | Default
+
+| `type`
+| Type of secret.
+| `Credentials`, `Keystore` or `Token`
+| true
+|
+
+| `secretName`
+| Name of referenced secret.
+| string
+| true
+|
+
+|=======================
+
+If type is `Credentials`, Secret` is expected to contain username and password credentials.
+These would be defined in `stringData/username` and `stringData/password` fields respectively.
+
+If type is `Keystore`, `Secret` is expected to contain base64 encoded data in `data/keystore.p12` field.
+Optional keystore password would be located in `stringData/password` field.
+
+If type is `Token`, `Secret` is expected to contain base64 encoded data in `stringData/token` field.
+
+
+[[infinispanstatus]]
+### `InfinispanStatus`
+
+`InfinispanStatus` is the most recent observed status of the `InfinispanSpec`. Read-only.
+
+This is a proposal. TODO: Please comment
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required
+
+| `nodes`
+| Status of the nodes.
+| []<<InfinispanNodeStatus>>
+| true
+
+| `summary`
+| Short description of the cluster status. Now summary is `Ready` if all the pod conditions of type `Ready` are true 
+| string
+| true
+
+|=======================
+
+
+[[nodes]]
+#### `InfinispanNodeStatus`
+
+`InfinispanNodeStatus` is the most recent observed status of a pod running `InfinispanSpec`.
+
+[options="header,footer"]
+|=======================
+| Field | Description | Scheme | Required
+
+| `name`
+| Name of the Pod.
+| string
+| true
+
+| `clusterMembers`
+| List of cluster members (in Infinispan CLI format)
+| string
+| true
+
+| `clusterSize`
+| Number of cluster members
+| int
+| true
+
+|=======================
+
+## Full Example
+
+.full-example.yaml
+[source,yaml]
+----
+apiVersion: infinispan.org/v1
+kind: Infinispan
+metadata:
+  name: full-example-infinispan
+spec:
+  evictionPolicy: Reject
+  image: jboss/infinispan-server:latest
+  replicationFactory: 3
+  size: 4
+  cluster:
+    encryption:
+      type: Symmetric
+      secret:
+        type: Keystore
+        secretName: cluster-encrypt-secret
+    secret:
+      type: Credentials
+      secretName: cluster-auth-secret
+  connector:
+    secret:
+      type: Credentials
+      secretName: connect-auth-secret
+  container:
+    jvmOptionsAppend: "-XX:NativeMemoryTracking=summary"
+    memory: 1Gi
+    storage: 2Gi
+  datasources:
+  - name: test-pg
+    driver: postgresql
+    secret:
+      type: Credentials
+      secretName: postgresql-secret
+  - name: test-mysql
+    driver: mysql
+    secret:
+      type: Credentials
+      secretName: mysql-secret
+  logging:
```

Comment:
Did you consider using yaml map for this?
```
loggingCategories:
  org.infinispan: trace
  org.jgroups: trace
```

Reply:
Good point, that's better.

Reply:
This is included in v2 of the design #77.